docs/addons/gmaestro.md

```diff
@@ -0,0 +1,67 @@
+# gMaestro add-on for Amazon EKS Blueprints
+
+The gMaestro Blueprints AddOn deploys the gMaestro Agent on Amazon EKS using the [eks-blueprints](https://github.com/aws-quickstart/cdk-eks-blueprints) [CDK](https://aws.amazon.com/cdk/) construct.
+
+
+## Prerequisite 
+Before using gMaestro, you need to:
+1. [Sign up](https://app.granulate.io/gMaestroSignup) to the gMaestro platform
+2. Download a sample YAML file - After signing up to gMaestro, navigate to the [Deploy](https://app.granulate.io/deploy) on the left-hand menu, fill in the required fields and click on "Generate Config File"
+
+![GmaestroGenerateConfigFile](./../assets/images/gmaestro-generate-config-file.png)
+
+![GmaestroConfigFile](./../assets/images/gmaestro-config-file.png)
+
+3. Create a secret (as a plaintext) in AWS Secrets Manager copy its value from the following place:
+   1. Navigate to Deployment section in the downloaded config file and use the value of `MAESTRO_CLIENT_ID` environment variable
+   2. Using AWS CLI:
+      ```bash
+      export MAESTRO_CLIENT_ID=<MAESTRO_CLIENT_ID>
+      export MAESTRO_SECRET_NAME=<MAESTRO_SECRET_NAME>
+      aws secretsmanager create-secret --name ${MAESTRO_SECRET_NAME} \
+          --description "Encrypted client ID for Granulate gMaestro" \
+          --secret-string "${MAESTRO_CLIENT_ID}"
+      ```
+   
+
+## Installation
+
+Using [npm](https://npmjs.org):
+
+```bash
+$ npm i @granulate/gmaestro-eks-blueprints-addon
+```
+
+## Usage
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import {GmaestroAddOn} from '@granulate/gmaestro-eks-blueprints-addon';
+
+const app = new cdk.App();
+
+const addOn = new GmaestroAddOn({
+        clientIdSecretName: "<secret name>", // MAESTRO_SECRET_NAME
```

Comment:
is clientIdSecretName a good parameter name? should it be more appropriate secretName? It is a bit confusing seems a merge between MAESTRO_SECRET_NAME and MAESTRO_CLIENT_ID